ckanext/tabledesigner/column_types.py

```diff
@@ -0,0 +1,306 @@
+
+
+@_standard_column('uuid')
+class UUIDColumn(ColumnType):
+    label = _('Universally unique identifier (UUID)')
+    description = _('A universally unique identifier as hexadecimal')
+    example = '213b972d-75c0-48b7-b14a-5a19eb58a1fa'
+    datastore_type = 'uuid'
+    table_schema_type = 'string'
+    table_schema_format = 'uuid'
+    _SQL_IS_EMPTY = "{value} IS NULL"
+
+
+@_standard_column('numeric')
+class NumericColumn(ColumnType):
+    label = _('Numeric')
+    description = _('Number with arbitrary precision (any number of '
+                    'digits before and after the decimal)')
+    example = '2.01'
+    datastore_type = 'numeric'
+    table_schema_type = 'number'
+    _SQL_IS_EMPTY = "{value} IS NULL"
+
+    def excel_validate_rule(self):
+        return 'NOT(ISNUMBER({_value_}))'
+
+
+@_standard_column('integer')
+class IntegerColumn(ColumnType):
+    label = _('Integer')
+    description = _('Whole numbers with no decimal')
+    example = '21'
+    datastore_type = 'int8'
+    table_schema_type = 'integer'
+    _SQL_IS_EMPTY = "{value} IS NULL"
+
+    def excel_validate_rule(self):
+        return 'NOT(IFERROR(INT({_value_})=VALUE({_value_}),FALSE))'
+
+
+@_standard_column('boolean')
+class BooleanColumn(ColumnType):
+    label = _('Boolean')
+    description = _('True or false values')
+    example = 'false'
+    datastore_type = 'boolean'
+    table_schema_type = 'boolean'
+    form_snippet = 'choice.html'
+    _SQL_IS_EMPTY = "{value} IS NULL"
+
+    def choices(self):
+        from ckan.plugins.toolkit import _
+        return {
+            'false': _('FALSE'),
+            'true': _('TRUE'),
+        }
+
+    def choice_value_key(self, value: bool | str) -> str:
+        """
+        convert bool to string for matching choice keys
+        """
+        return 'true' if value else 'false' if isinstance(
+            value, bool) else value
+
+    def excel_validate_rule(self):
+        return 'AND({_value_}<>TRUE,{_value_}<>FALSE)'
+
+
+@_standard_column('json')
+class JSONColumn(ColumnType):
+    label = _('JSON')
+    description = _('A JSON object')
+    example = '{"key": "value"}'
+    datastore_type = 'json'
+    table_schema_type = 'object'
+    _SQL_IS_EMPTY = "{value} IS NULL OR {value}::jsonb = 'null'::jsonb"
+
+
+@_standard_column('date')
+class DateColumn(ColumnType):
+    label = _('Date')
+    description = _('Date without time of day')
+    example = '2024-01-01'
+    datastore_type = 'date'
+    table_schema_type = 'date'
+    table_schema_format = 'fmt:YYYY-MM-DD'
+    html_input_type = 'date'
+    excel_format = 'yyyy-mm-dd'
+    _SQL_IS_EMPTY = "{value} IS NULL"
+
+    def excel_validate_rule(self):
+        return 'NOT(ISNUMBER({_value_}+0))'
+
+
+@_standard_column('timestamp')
+class TimestampColumn(ColumnType):
+    label = _('Timestamp')
+    description = _('Date and time without time zone')
+    example = '2024-01-01 12:00:00'
+    datastore_type = 'timestamp'
+    table_schema_type = 'time'
```

Comment:
The [Frictionless "time" field](https://framework.frictionlessdata.io/docs/fields/time.html) is a time without a date, so you may want to use "timestamp" for the `table_schema_type` here if you are later going to use this variable as the Frictionless equivalent...